src/snowflake/snowpark/session.py

```diff
@@ -376,6 +378,7 @@ def __init__(
         )
         self._conf = self.RuntimeConfig(self, options or {})
         self._tmpdir_handler: Optional[tempfile.TemporaryDirectory] = None
+        self._runtime_version: str = None
```

Comment:
nit: probably name this `udf_server_runtime_version`/`runtime_version_from_requirement` or something like this so that we don't confuse it with the current runtime version of the client.